### ISSUE TITLE
Calling refreshFormData on repeat fields clears them

### ISSUE THREAD

**Issue by euoia** [laravel-filament/filament#14222]
### Package

filament/filament

### Package Version

v3.2.110

### Laravel Version

v11.22.0

### Livewire Version

v3.5.6

### PHP Version

PHP 8.3.8

### Problem description

I called `refreshFormData` with fields fields that are defined using `Repeater::make` and it cleared them.

### Expected behavior

I expected the fields to be updated with their latest values.

### Steps to reproduce

1. Create a simple Repeater field in your Resource:
```
                Repeater::make('comments')
                    ->simple(
                        TextInput::make('comment'),
                    ),
```

2. Add an action to `getHeaderActions` in the EditRecord:
```
            Actions\Action::make('update comments')
                ->action(function () {
                    // Modify comments
                    $this->record->comments = ['foo', 'bar'];
                    $this->record->save();

                    $this->refreshFormData([
                        'comments',
                    ]);
                }),
```

3. Observe that the comments repeater fields are cleared instead of updated with new values.

### Reproduction repository (issue will be closed if this is not valid)

https://github.com/cloudshipco/filamentphp-repeater-refresh-bug

### Relevant log output

```shell
n/a
```

**Comment by yuters** [comment]
In that scenario I would typically use `$this->fillForm()` instead of `$this->refreshFormData()`.

Otherwise, looking at your form schema, the refreshFormData would typically look for an array of comments with each having a `comment` key corresponding to the name of the TextInput. So this change should work:

```diff
- $this->record->comments = ['foo', 'bar'];
+ $this->record->comments = [['comment' => 'foo'], ['comment' => 'bar']];
```

**Comment by zepfietje** [comment]
@yuters, simple repeaters do actually use a flat array to store items, so this might in fact be a bug.

**Comment by yuters** [comment]
@zepfietje yep could probably be improved for simple repeaters. Just suggesting some workarounds.

**Comment by danharrin** [comment]
Should be fixed by #15514
